### PR TITLE
Wire up js langs

### PR DESCRIPTION
**Pull request by tpetr**
This PR is quite large, but should be digestible if you break it down by the first 3 commits:

1. Some final massaging of the ocaml <> JS bridge + implement a couple languages
2. Implement the rest of the languages
3. Update the workflow to build & test the JS engine + all languages

### CODE REVIEW

**Comment by aryx** [comment]
CI seems blocked ...

**Comment by tpetr** [comment]
Yep, investigating

**Comment by aryx** [comment]
I foresee we will never get a full list of green checks anymore

**Comment by tpetr** [comment]
Seemed to be a transient issue with cache restore. If we see this happen often we can disable testing the js langs outside of the nightly run and releases

**Comment by aryx** [comment]
I want to release 1.19.0 today before merging this

File: .github/workflows/build-test-javascript.yaml [hunk 1]
```diff
@@ -19,42 +18,79 @@ jobs:
       - uses: actions/checkout@v3
         with:
           submodules: true
+      - name: Set up tree-sitter
+        run: |
+          (cd libs/ocaml-tree-sitter-core && ./configure && ./scripts/install-tree-sitter-lib)
+      - name: Cache git checkout
```
Comment (aryx) [inline .github/workflows/build-test-javascript.yaml:24]: Do we need this? Does it improve significantly the build time?

**Review by aryx — APPROVED** [review]

File: .github/workflows/build-test-javascript.yaml (hunk 1 above)
Comment (tpetr) [reply .github/workflows/build-test-javascript.yaml:24]: 40+ seconds to do a full clone vs. 4s to restore repository from cache

**Pull request merged**